dcpy/test/lifecycle/assemble/test_assemble.py

```diff
@@ -31,3 +35,41 @@ def test_unpack_package(metadata, tmp_path, package_path):
     # Check that files are unpacked correctly
     assert (dataset_files_path / "data_file.xlsx").exists()
     assert (attachments_path / "readme.pdf").exists()
+
+
+@pytest.fixture
+def colp_package_path(resources_path: Path):
+    return resources_path / "product_metadata" / "colp_single_feature_package"
+
+
+@mock.patch("dcpy.lifecycle.distribute.bytes.pull_destination_files")
+def test_assemble_from_bytes(pull_destination_files_mock, tmp_path, colp_package_path):
+    OTI_DD_FILENAME = "oti_dd.xlsx"
+    MOCK_PULLED_PACKAGE_PATH = tmp_path
+
+    pull_destination_files_mock.side_effect = lambda *args, **kwargs: shutil.copytree(
+        colp_package_path, MOCK_PULLED_PACKAGE_PATH, dirs_exist_ok=True
+    )
+
+    # Add an oti_xlsx file type, which we'll then have to generate
+    metadata = md.Metadata.from_path(
+        colp_package_path / "metadata.yml", template_vars={"version": "24b"}
+    )
+    metadata.files.append(
+        md.FileAndOverrides(
+            file=md.File(
+                id="oti_xlsx", filename=OTI_DD_FILENAME, type=OTI_METADATA_FILE_TYPE
+            )
+        )
+    )
+
+    assemble.assemble_dataset_from_bytes(
+        metadata, destination_id="socrata", out_path=tmp_path
+    )
+
+    attachments_path = MOCK_PULLED_PACKAGE_PATH / "attachments"
+    assert attachments_path.exists(), "Sanity check that the mock side_effect works"
```

Comment:
is this assertion moot after `assemble.assemble_dataset_from_bytes` succeeds and covered by the next assertion?

Reply:
yeah, I left it in because that's probably the first thing that's gonna fail when we mess with these tests again. Could effectively remove it, but I tend to like those little checkpoints